avar2.md

```diff
@@ -30,27 +30,36 @@ Use cases include:
 
 The *DeltaSetIndexMap* and *ItemVariationStore* formats are given in [OpenType Font Variations Common Table Formats](https://docs.microsoft.com/en-us/typography/opentype/spec/otvarcommonformats).
 
 # Processing
 
-Processing of axis values in an `avar` version 2 table happens in 3 stages for a given instance:
+Processing of axis values in an `avar` version 2 table happens in 3 stages for an instance determined by a set of user coordinates:
+
+1. Convert user coordinates of each axis to initial normalized coordinates in the range [-1.0, 1.0]. (This step always happens in OpenType Variations.)
+2. Remap initial normalized coordinates of each axis via the *axisSegmentMap*s, providing intermediate coordinates also in the range [-1.0, 1.0]. (This step is the usual `avar` version 1 remapping.)
+3. Calculate interpolated deltas for each axis and add them to intermediate coordinates, providing final coordinates that are clamped to the range [-1.0, 1.0].
 
-1. Initial normalization to convert user coordinates of each axis to initial normalized coordinates in the range [-1.0, 1.0].
-2. Remap initial normalized coordinates of each axis via the *axisSegmentMap*s of `avar` version 1, providing intermediate coordinates also in the range [-1.0, 1.0].
-3. Calculate interpolated deltas for each axis via `avar` version 2 and add them to intermediate coordinates, providing final coordinates that are clamped to the range [-1,1].
+In more detail, step 3 proceeds as follows, using existing methods of processing *ItemVariationStore* structures:
 
-In more detail, step 3 proceeds as follows. Considering the coordinates provided by step 2, a scalar is determined for each *variationRegion* by the usual variation interpolation algorithm. Then, the *deltaSet*s of each *ItemVariationData* are processed such that each delta value is multiplied by its associated scalar. Summing those products gives an interpolated delta to add to a particular axis coordinate, the axis index being defined in `axisIndexMap`. After the summing operations for all *ItemVariationData* structures is complete, axis values are clamped to to the range [-1.0, 1.0], giving final axis coordinates.
+* Considering the coordinates provided by step 2, calculate a scalar for each *variationRegion* in the *ItemVariationStore* by the usual variation interpolation algorithm.
+* For each axis:
+  * Look up its outer and inner indices in `axisIndexMap`.
+  * The outer value identifies an *ItemVariationData*.
+  * The inner value identifies a *deltaSet* within the *ItemVariationData*.
+  * The *ItemVariationData* contains an array of *variationRegion* indices, implying an array of indices to scalars.
+  * Multiply each scalar by the delta value at the same index, yielding an array of products. (We can do this since the array of region indices has the same length as the *deltaSet*.)
+  * Take the sum of those products to obtain the interpolated delta value.
```

Comment:
The substeps up to here are internals of *ItemVariationStore* and should not be explained. Really the algorithm for this table is 3 lines and I prefer to keep it that way.

Reply:
My issue is probably with the lack of a good algorithm description in the spec for *ItemVariationStore*.